File: lib/src/db/entities/session.entity.ts
import { BaseEntity } from "../../common/base/base.entity";
import { ISession } from "@traceo/types";
import { Column, Entity, PrimaryGeneratedColumn } from "typeorm";

@Entity()
export class Session extends BaseEntity implements ISession {
  @PrimaryGeneratedColumn("uuid")
  id?: string;

  @Column({
    type: "varchar"
  })
  sessionID: string;

  @Column({
    type: "varchar"
  })
  userID: string;

  @Column({
    type: "varchar",
    nullable: false
  })
  userName: string;

  @Column({
    type: "varchar"
  })
  userIP: string;

  @Column({
    type: "bigint",
    nullable: false
  })
  expiryAt: number;

  @Column({
    type: "bigint",
    nullable: true
  })
  revokedAt: number;
}

File: server/traceo-common/traceo-common-jpa/src/main/java/org/traceo/common/jpa/entities/SessionEntity.java
package org.traceo.common.jpa.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "traceo_user_session")
@Getter
@Setter
public class SessionEntity extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(unique = true, name = "session_id")
    private String sessionID;

    @Column(nullable = false, name = "user_id")
    private String userID;

    @Column(nullable = false)
    private String username;

    @Column(nullable = false, name = "user_ip")
    private String userIP;

    @Column(nullable = false, name = "expiry_at")
    private long expiryAt;

    @Column
    private long revokedAt;
}
